packages/flutter_tools/lib/src/build_system/file_hash_store.dart

```diff
@@ -113,23 +124,27 @@ class FileHashStore {
   /// Persist file hashes to disk.
   void persist() {
     printTrace('Persisting file store');
-    final File file = _cacheFile;
-    if (!file.existsSync()) {
-      file.createSync(recursive: true);
+    final File cacheFile = fileSystem.file(_cachePath);
+    if (!cacheFile.existsSync()) {
+      cacheFile.createSync(recursive: true);
     }
     final List<FileHash> fileHashes = <FileHash>[];
     for (MapEntry<String, String> entry in currentHashes.entries) {
-      previousHashes[entry.key] = entry.value;
-    }
-    for (MapEntry<String, String> entry in previousHashes.entries) {
       fileHashes.add(FileHash(entry.key, entry.value));
     }
     final FileStorage fileStorage = FileStorage(
       _kVersion,
       fileHashes,
     );
     final Uint8List buffer = fileStorage.toBuffer();
-    file.writeAsBytesSync(buffer);
+    try {
+      cacheFile.writeAsBytesSync(buffer);
+    } on FileSystemException catch (err) {
+      printError(
+        'Failed to persist file store at ${cacheFile.path} due to $err.\n'
+        'Build artifacts will not be cached.'
```

Comment:
Ditto comment here about suggesting to delete the file manually.

Reply:
Done